components/external_link/index.tsx

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable mattermost/use-external-link */
+
+import React from 'react';
+import {useSelector} from 'react-redux';
+
+import {trackEvent} from 'actions/telemetry_actions';
+import {getConfig, getLicense} from 'mattermost-redux/selectors/entities/general';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/common';
+
+type ExternalLinkQueryParams = {
+    utm_source?: string;
+    utm_medium?: string;
+    utm_campaign?: string;
+    utm_content?: string;
+    userId?: string;
+}
+
+type Props = React.AnchorHTMLAttributes<HTMLAnchorElement> & {
+    href?: string;
+    target?: string;
+    rel?: string;
+    onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+    queryParams?: ExternalLinkQueryParams;
+    location?: string;
+    children: React.ReactNode;
+};
+
+export default function ExternalLink(props: Props) {
+    const userId = useSelector(getCurrentUserId);
+    const config = useSelector(getConfig);
+    const license = useSelector(getLicense);
+    let href = props.href;
+    let queryParams = {};
+    if (href?.includes('mattermost.com')) {
+        // encode this stuff so it's not so transparent to the user?
+        const existingURLSearchParams = new URL(href).searchParams;
+        const existingQueryParamsObj = Object.fromEntries(existingURLSearchParams.entries());
+        queryParams = {
+            utm_source: 'mattermost',
+            utm_medium: license.Cloud === 'true' ? 'in-product-cloud' : 'in-product',
+            utm_content: props.location || '',
+            uid: userId,
```

Comment:
Is user id necessary?

Reply:
Kevin mentioned it in lieu of PII. I think it would be useful for the data team since we'd be able to map this value to the `user_actual_id` from rudder events.